### PR TITLE
Effect Icepiece OK

### PR DESCRIPTION
**Pull request by hensldm**
More or less copy paste from OoT.

### CODE REVIEW

File: src/overlays/effects/ovl_Effect_Ss_Ice_Piece/z_eff_ss_ice_piece.c [hunk 1]
```diff
@@ -1,27 +1,84 @@
...
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s32 pad;
+    f32 scale = this->rScale * 0.01f;
+    u32 frames = globalCtx->state.frames;
+    f32 alpha;
+
+    OPEN_DISPS(gfxCtx);
+
+    if ((this->rLifespan > 0) && (this->life < (this->rLifespan >> 1))) {
+        alpha = ((this->life * 2.0f) / this->rLifespan);
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Ice_Piece/z_eff_ss_ice_piece.c:57]:
```suggestion
        alpha = (this->life * 2.0f) / this->rLifespan;
```
I don't think this bracket is needed

File: src/overlays/effects/ovl_Effect_Ss_Ice_Piece/z_eff_ss_ice_piece.c [hunk 2]
```diff
@@ -1,27 +1,84 @@
...
+    Matrix_Translate(this->pos.x, this->pos.y, this->pos.z, MTXMODE_NEW);
+    Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    Matrix_RotateY(this->rYaw, MTXMODE_APPLY);
+    Matrix_RotateXS(this->rPitch, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    func_8012C2DC(globalCtx->state.gfxCtx);
+    gDPSetEnvColor(POLY_XLU_DISP++, 0, 50, 100, (s32)alpha & 0xFF);
+    func_800BCC68(&this->pos, globalCtx);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, (1 * frames) % 256, 0x20, 0x10, 1, 0, (2 * frames) % 256,
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Ice_Piece/z_eff_ss_ice_piece.c:72]:
```suggestion
               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, frames % 256, 0x20, 0x10, 1, 0, (2 * frames) % 256,
```
Do you really need the `1 *`?

**Review by isghj5 — APPROVED** [review]
Do we know yet what spawns this effect? I searched /overlays but nothing came up

**Comment by hensldm** [comment]
> Do we know yet what spawns this effect? I searched /overlays but nothing came up
Looks like `Player` uses `EffectSsIcePiece_SpawnBurst`, but nothing else from what I can see.

**Review by engineer124 — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
n-ICE work

**Review by EllipticEllipsis — APPROVED** [review]
Possibly the `(1 * )` is better if it turns out there's a good way to macro this.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**